utils/funcexp/func_rand.cpp

```diff
@@ -70,6 +71,37 @@ double Func_rand::getDoubleVal(rowgroup::Row& row,
     // NOTE: this function needs to use 32bit ints otherwise it will break for negative values
     uint32_t seedParm = 0;
 
+    // Still on first row, multiple rands exist in statement. Need an additional seed.
+    if(fFirstRow == row.getData())
+    {
+        fSeedSet = false;
+        fSeedIndex += 1;
+    }
+    else if (fFirstRow == 0)
+    {
+        // Store first row
+        fFirstRow = row.getData();
+        fSeedIndex = 0;
+    }
+    else
+    {
+        // No longer on first row. Disable additional seeding
+        if (!fMultipleSeedsSet)
+        {
+            fMultipleSeedsSet = true;
+            fSeedIndex = 0;
+        }
+        else
+        {
+            fSeedIndex += 1;
```

Comment:
I wonder if we need to take a lock working with Func_rand intsances from multiple threads or they are separate?